webapp/channels/src/components/save_button.tsx

```diff
@@ -17,65 +17,55 @@ type Props = {
     extraClasses?: string;
 }
 
-export default class SaveButton extends React.PureComponent<Props> {
-    public static defaultProps: Partial<Props> = {
-        btnClass: '',
-        defaultMessage: (
-            <FormattedMessage
-                id='save_button.save'
-                defaultMessage='Save'
-            />
-        ),
-        disabled: false,
-        extraClasses: '',
-        savingMessage: (
-            <FormattedMessage
-                id='save_button.saving'
-                defaultMessage='Saving'
-            />
-        ),
-    };
-
-    public render() {
-        const {
-            saving,
-            disabled,
-            savingMessage,
-            defaultMessage,
-            btnClass,
-            extraClasses,
-            ...props
-        } = this.props;
-
-        let className = 'btn';
-        if (!btnClass) {
-            className += ' btn-primary';
-        }
+const SaveButton = ({
+    disabled = false,
+    btnClass = '',
+    extraClasses = '',
+    savingMessage = (
+        <FormattedMessage
+            id='save_button.saving'
+            defaultMessage='Saving'
+        />
+    ),
+    defaultMessage = (
+        <FormattedMessage
+            id='save_button.save'
+            defaultMessage='Save'
+        />
+    ),
+    saving,
+    ...props
+}: Props) => {
+    let className = 'btn';
+    if (!btnClass) {
+        className += ' btn-primary';
+    }
 
-        if (!disabled || saving) {
-            className += ' ' + btnClass;
-        }
+    if (!disabled || saving) {
+        className += ' ' + btnClass;
+    }
 
-        if (extraClasses) {
-            className += ' ' + extraClasses;
-        }
+    if (extraClasses) {
+        className += ' ' + extraClasses;
+    }
```

Comment:
A bit out of scope, but can we use the `classnames` utility function to form this class?